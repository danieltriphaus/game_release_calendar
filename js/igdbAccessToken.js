const axios = require("axios");
/**
 * @module igdb/igdbAccessToken
 */

/** @type {IGDBAccessToken} */
let accessToken = {
    access_token: "",
    expires_in: 0,
    token_type: "",
};


module.exports = {
    /**
     * @async
     * @returns {Promise<IGDBAccessToken>}
     */
    getIgdbAccessToken: async () => {
        if (accessToken.access_token.length === 0) {
            const response = await axios
                .post(
                    "https://id.twitch.tv/oauth2/token?client_id=" +
                    process.env.IGDB_API_CLIENT_ID +
                    "&client_secret=" +
                    process.env.IGDB_API_CLIENT_SECRET +
                    "&grant_type=client_credentials",
                )
                .catch((error) => {
                    console.warn("Warning: igdb Access Token could not be obtained");
                    console.error(error);
                });

            if (response) {
                accessToken = response.data;
                console.log("igdb Access Token sucessfully obtained " + accessToken.access_token);
                return response.data;
            }
        } else {
            return accessToken;
        }
    },


    clearLocalAccessToken: () => {
        accessToken = {
            access_token: "",
            expires_in: 0,
            token_type: "",
        };
    }
};
