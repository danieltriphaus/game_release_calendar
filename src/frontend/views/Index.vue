<template>
    <div class="container">
        <div v-if="isAuthenticated">
            <router-link to="/calendar">Kalendar generieren</router-link>
        </div>

        <div v-if="!isAuthenticated" class="row mt-4">
            <form name="api-validate" @submit.prevent="validateApiKey">
                <div class="col-md-10">
                    <input type="password" id="api-key" name="api-key" class="form-control" placeholder="API Key" v-model="apiKey">
                </div>
                <div class="col-md-2">
                    <input type="submit" class="btn btn-outline-primary" data-testid="api-validate">
                </div>
            </form>
        </div>
        
        <div v-else class="row mt-4">
            <div class="col">
                <input type="search" v-debounce:500ms="searchGames" v-model="searchQuery" id="search-games" name="search-games" class="form-control" placeholder="Suche Game">
            </div>
        </div>

        <div class="results mt-2" v-for="result in searchResults" :key="result.id">
            <div class="row">
                <div class="col result mt-2" :data-testid="'result-' + result.id" @click="addGame(result.id)">
                    <img :src="result.cover.url.replace('thumb', 'cover_small')" class="game-cover">
                    <div class="game-info">
                        <h5>{{ result.name }}</h5>
                        <h6>{{ result.involved_companies.find((involved_company) => involved_company.developer === true).company.name }}</h6>
                    </div>
                </div>
            </div>
        </div>
    </div>
</template>

<script>
//ToDo: Refactor into Components
import axios from "axios";
import { getDirective } from "vue-debounce";

export default {
  name: "IndexView",
  directives: {
    debounce: getDirective(3)
  },
  components: {
  },
  data() {
    return { 
      isAuthenticated: false,
      apiKey: "",
      searchQuery: "",
      searchResults: [],
    }
  },
  async mounted() {
    const response = await axios.get("/api/access").catch((error) => {
      if (error.response && error.response.status === 401) {
        this.isAuthenticated = false;
      }
    });
    if (response && response.status === 200) {

      this.isAuthenticated = true;
    }
  },
  methods: {
    async validateApiKey() {
      const form = new FormData();
      form.append("apiKey", this.apiKey);
      const response = await axios.post("/api/access", form).catch((error) => {console.log(error)});
      if (response) {
        this.isAuthenticated = true;
      }
    },
    async searchGames() {
      const response = await axios.get("/api/game/search", { params: { q: this.searchQuery }});
      this.searchResults = response.data;
    },
    async addGame(id) {
      await axios.post("/api/user/y1xx/games", [id]);
    }
  },
  computed: {
  }
};
</script>

<style scoped>
    .result {
        display: flex;
        cursor: pointer;
        padding: 2px;
    }

    .result:hover {
        background-color: rgb(211, 209, 209);
    }

    .game-cover {   
        height: 100px;
    }

    .game-info {
        margin-top: 10px;
        padding-left: 10px;
    }
</style>