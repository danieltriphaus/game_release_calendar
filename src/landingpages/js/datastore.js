const { Datastore } = require("@google-cloud/datastore");

/**
 * Get game lists which contain a specific game id
 * @async

 * @param {UserID} userId
 * @param {Boolean} withGames
 * @returns {Promise<Array>}
 */
module.exports = {
    getAllGameLists: async (userId) => {
        const datastore = new Datastore({
            projectId: "grc-6732",
            keyFilename: "./event_credentials.json",
        });
        const query = datastore.createQuery("game_list");
        if (userId) {
            query.hasAncestor(datastore.key(["user", userId]));
        }
        const [gameLists] = await datastore.runQuery(query);

        return gameLists;
    },

    /**
     * convertFromDatastoreResult
     * @param {Array | undefined} result
     * @returns {Array | undefined}
     */
    convertFromDatastoreResult: (result) => {
        return result
            ? result.map((datastoreResult) => {
                return { id: datastoreResult[Datastore.KEY].name ? datastoreResult[Datastore.KEY].name : datastoreResult[Datastore.KEY].id, ...datastoreResult };
            })
            : [];

    },

    filterEventLists: (gameLists) => {
        return gameLists.filter((gameList) => {
            return gameList[Datastore.KEY].name !== "default" && gameList[Datastore.KEY].name !== "archive";
        });
    },
};