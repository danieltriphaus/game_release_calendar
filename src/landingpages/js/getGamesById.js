const axios = require("axios");
const { getIgdbAccessToken } = require("../js/igdbAccessToken");
const { getAllGameLists, filterEventLists, convertFromDatastoreResult } = require("../js/datastore");

const displayFields = [
    "name",
    "first_release_date",
    "url",
    "release_dates.category",
    "release_dates.human",
    "release_dates.date",
    "release_dates.platform.id",
    "release_dates.platform.abbreviation",
    "release_dates.platform.alternative_name",
    "release_dates.platform.name",
    "cover.height",
    "cover.width",
    "cover.url",
    "involved_companies.developer",
    "involved_companies.company.name",
];

const query = "fields {fields}; where id = ({gameIds}); limit 500;";

async function getGamesById(gameIds, accessToken) {
    if (!gameIds || gameIds.length === 0) {
        return [];
    }

    const gameIdsString = gameIds.join(",");

    const finalQuery = query.replace("{fields}", displayFields.join(","));

    const response = await axios.post("https://api.igdb.com/v4/games", finalQuery.replace("{gameIds}", gameIdsString), {
        headers: {
            Accept: "application/json",
            "Content-Type": "text/plain",
            "Client-ID": process.env.IGDB_API_CLIENT_ID,
            Authorization: "Bearer " + accessToken,
        },
    });

    return response.data;
}

let eventListsGames = (async () => {
    const accessToken = await getIgdbAccessToken();
    const gameLists = await getAllGameLists(process.env.EVENT_USER_ID);
    const eventLists = convertFromDatastoreResult(filterEventLists(gameLists));

    const gameIds = eventLists.reduce((accumulator, currentValue) => {
        return [...accumulator, ...currentValue.games.map((game) => game.id)];
    }, []);

    const games = await getGamesById(gameIds, accessToken.access_token);

    return eventListsGames = eventLists.map((eventList) => {
        return {
            ...eventList,
            games: eventList.games.map((game) => {
                return games.find((gameData) => gameData.id === game.id);
            }),
        };
    });
})();


module.exports = {
    getGames: () => eventListsGames,
};

