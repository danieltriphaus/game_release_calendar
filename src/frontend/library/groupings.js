import { getSelectedReleaseDate } from "./releaseDate.js";

/**
 * @typedef {Object} Category
 * @property {string} id
 * @property {number} key
 * @property {string} heading
 *
 * @typedef {Category[]} Grouping
 */

/**
 * @type {string}
 */
let defaultGrouping = "released_unreleased";

/**
 * @param {*} gameAssignments
 * @returns {Grouping}
 */
export const getDefaultGrouping = () => {
    return defaultGrouping;
};

/**
 * @param {string} grouping
 */
export const setDefaultGrouping = (grouping) => {
    defaultGrouping = grouping;
};

export const getCurrentCategories = (currentGrouping, sortedGames) => {
    const categories = config.groupings[currentGrouping];

    const initialValue = {};

    categories.forEach((category) => {
        initialValue[category.id] = [];
    });

    const categoriesMap = sortedGames.reduce(config.sorting[currentGrouping](), initialValue);

    categories.forEach((category) => {
        category.games = categoriesMap[category.id];
    });

    return categories.filter((category) => category.games.length > 0);
};


function releasedUnreleased() {
    return (categoriesMap, game) => {
        if (!game.release_dates || !game.release_dates.length) {
            categoriesMap["unreleased-games"].push(game);
        } else {
            const selectedReleaseDate = getSelectedReleaseDate(game);
            const releaseDate = new Date(selectedReleaseDate.date * 1000);

            if (releaseDate <= new Date()) {
                categoriesMap["released-games"].push(game);
            } else {
                categoriesMap["unreleased-games"].push(game);
            }
        }
        return categoriesMap;
    };
}

function monthly() {
    const currentYear = new Date().getFullYear();
    return (categoriesMap, game) => {
        const selectedReleaseDate = getSelectedReleaseDate(game);
        const releaseDate = new Date(selectedReleaseDate.date * 1000);

        if (selectedReleaseDate.category === 2 && releaseDate.getFullYear() === currentYear) {
            const categoryID = `year-${releaseDate.getFullYear()}`;
            categoriesMap[categoryID].push(game);
        } else if (selectedReleaseDate.category === 7) {
            categoriesMap["tbd"].push(game);
        } else if (releaseDate.getFullYear() < currentYear) {
            categoriesMap["past"].push(game);
        } else if (releaseDate.getFullYear() > currentYear) {
            categoriesMap["future"].push(game);
        } else {
            const month = releaseDate.toLocaleString("en", { month: "long" }).toLowerCase();
            categoriesMap[month].push(game);
        }
        return categoriesMap;
    };
}

export const config = {
    "default": "released_unreleased",
    /**
	 * @type {Grouping[]}
	 */
    sorting: {
        "released_unreleased": releasedUnreleased,
        "months": monthly,
    },
    "groupings": {
        /**
		 * @type {Grouping}
		 */
        "released_unreleased": [
            {
                "id": "released-games",
                "key": 1,
                "heading": "Released Games",
            },
            {
                "id": "unreleased-games",
                "key": 2,
                "heading": "Unreleased Games",
            },
        ],
        /**
		 * @type {Grouping}
		 */
        "months": [
            {
                id: "past",
                key: 0,
                heading: "Past",
            },
            {
                "id": "january",
                "key": 1,
                "heading": "January",
            },
            {
                "id": "february",
                "key": 2,
                "heading": "February",
            },
            {
                "id": "march",
                "key": 3,
                "heading": "March",
            },
            {
                "id": "april",
                "key": 4,
                "heading": "April",
            },
            {
                "id": "may",
                "key": 5,
                "heading": "May",
            },
            {
                "id": "june",
                "key": 6,
                "heading": "June",
            },
            {
                "id": "july",
                "key": 7,
                "heading": "July",
            },
            {
                "id": "august",
                "key": 8,
                "heading": "August",
            },
            {
                "id": "september",
                "key": 9,
                "heading": "September",
            },
            {
                "id": "october",
                "key": 10,
                "heading": "October",
            },
            {
                "id": "november",
                "key": 11,
                "heading": "November",
            },
            {
                "id": "december",
                "key": 12,
                "heading": "December",
            },
            {
                "id": "year-" + new Date().getFullYear(),
                "key": 13,
                "heading": new Date().getFullYear()
                    .toString(),
            },
            {
                "id": "future",
                "key": 14,
                "heading": new Date(new Date().getFullYear() + 1, 1, 1).getFullYear()
                    .toString() + " and beyond",
            },
            {
                id: "tbd",
                key: 15,
                heading: "TBD",
            },
        ],
    },
};
