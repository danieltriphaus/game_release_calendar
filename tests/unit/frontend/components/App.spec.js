import { mount, flushPromises } from "@vue/test-utils";
import Index from "@/frontend/views/Index.vue";
import axios from "axios";
import testData from "./GameList/testData.json";
import { BootstrapVue3 } from "bootstrap-vue-3";
import { setDefaultGrouping } from "@/frontend/library/groupings.js";

jest.mock("axios");

it("renders correctly with released unreleased grouping", async () => {

    axios.get.mockResolvedValueOnce({ data: [testData.futureGame, testData.nextYearGame, testData.juneGame, testData.releasedGame] });

    const wrapper = await mount(Index, {
        global: {
            plugins: [BootstrapVue3],
            provide: {
                isAuthenticated: true,
                user: { value: { id: "y1xx" }, id: "y1xx" },
            },
        },
    });

    await flushPromises();

    expect(wrapper.html()).toMatchSnapshot();
});

it("renders correctly with monthly grouping", async () => {
    setDefaultGrouping("months");
    axios.get.mockResolvedValueOnce({ data: [testData.futureGame, testData.nextYearGame, testData.juneGame, testData.releasedGame] });

    const wrapper = await mount(Index, {
        global: {
            plugins: [BootstrapVue3],
            provide: {
                isAuthenticated: true,
                user: { value: { id: "y1xx" }, id: "y1xx" },
            },
        },
    });

    await flushPromises();

    expect(wrapper.html()).toMatchSnapshot();
});