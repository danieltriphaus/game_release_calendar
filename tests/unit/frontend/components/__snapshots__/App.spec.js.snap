// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with monthly grouping 1`] = `
"<div class=\\"b-overlay-wrap position-relative\\"><span><div><div><div class=\\"row mt-2\\"><div class=\\"col\\"><div class=\\"input-group\\" role=\\"group\\"><!----><input type=\\"search\\" data-cy=\\"search-games\\" name=\\"search-games\\" class=\\"form-control\\" placeholder=\\"Search Games\\"><div class=\\"btn-group user-menu\\"><button class=\\"btn btn-light dropdown-toggle dropdown-toggle-no-caret\\" type=\\"button\\" id=\\"user-menu\\" data-bs-toggle=\\"dropdown\\" aria-expanded=\\"false\\"> <i class=\\"bi-person-circle\\"></i></button><!----><ul class=\\"dropdown-menu\\" aria-labelledby=\\"user-menu\\" role=\\"menu\\"><li role=\\"presentation\\"><button class=\\"dropdown-item text-danger\\" type=\\"button\\" target=\\"_self\\"><i class=\\"bi-door-open me-2\\"></i>Sign Out </button></li></ul></div></div><!--v-if--></div></div></div><div><ul class=\\"list-actions\\" data-test=\\"list-actions\\"><li id=\\"temporary_game\\" class=\\"\\">Game not found?</li><li id=\\"calendar\\" class=\\"\\">Calendar</li><li id=\\"archive\\" class=\\"\\">Archive</li><li id=\\"grouping\\" class=\\"\\">Grouping</li></ul><div class=\\"list-actions-content\\"><!--v-if--><!--v-if--><!--v-if--><!--v-if--></div></div></div><div><!--v-if--><div class=\\"list-category\\"><div class=\\"list-category-heading\\"><h5 class=\\"list-heading\\">February</h5><i class=\\"list-icon bi-caret-down-fill\\"></i></div><div id=\\"february\\" class=\\"collapse\\"><div class=\\"row\\" data-cy=\\"february\\"><div id=\\"game-194740\\" class=\\"col game\\" data-testid=\\"game\\" data-cy=\\"game\\"><img loading=\\"lazy\\" src=\\"//images.igdb.com/igdb/image/upload/t_cover_small/co4kxj.jpg\\" class=\\"game-cover\\" data-testid=\\"game-194740-cover\\"><div class=\\"game-info\\"><h5 data-cy=\\"game-title\\">Deliver Us Mars</h5><h6>KeokeN Interactive</h6><h6 data-testid=\\"release-date\\" data-cy=\\"release-date\\" class=\\"release-date\\">02/02/2023</h6></div><div class=\\"platforms\\" data-cy=\\"platforms-194740\\"><span class=\\"platform platform-selector PC\\" title=\\"PC (Microsoft Windows)\\">PC</span><span class=\\"platform platform-selector XONE\\" title=\\"Xbox One\\">XONE</span><span class=\\"platform platform-selector XSX\\" title=\\"Xbox Series X|S\\">XSX</span><span class=\\"platform platform-selector PS4\\" title=\\"PlayStation 4\\">PS4</span><span class=\\"platform platform-selector PS5\\" title=\\"PlayStation 5\\">PS5</span></div>
<div class=\\"game-actions\\"><a href=\\"https://www.igdb.com/games/deliver-us-mars\\" target=\\"_blank\\" class=\\"game-details btn btn-igdb\\"><i class=\\"bi bi-info-square\\"></i></a><br><button type=\\"button\\" class=\\"btn btn-outline-danger\\" data-cy=\\"delete-game\\"><i class=\\"bi bi-trash-fill\\"></i></button></div>
</div>
</div>
</div>
</div>
<div class=\\"list-category\\">
  <div class=\\"list-category-heading\\">
    <h5 class=\\"list-heading\\">June</h5><i class=\\"list-icon bi-caret-down-fill\\"></i>
  </div>
  <div id=\\"june\\" class=\\"collapse\\">
    <div class=\\"row\\" data-cy=\\"june\\">
      <div id=\\"game-191692\\" class=\\"col game\\" data-testid=\\"game\\" data-cy=\\"game\\"><img loading=\\"lazy\\" src=\\"//images.igdb.com/igdb/image/upload/t_cover_small/co5vst.jpg\\" class=\\"game-cover\\" data-testid=\\"game-191692-cover\\">
        <div class=\\"game-info\\">
          <h5 data-cy=\\"game-title\\">Street Fighter 6</h5>
          <h6>Capcom</h6>
          <h6 data-testid=\\"release-date\\" data-cy=\\"release-date\\" class=\\"release-date\\">06/02/2023</h6>
        </div>
        <div class=\\"platforms\\" data-cy=\\"platforms-191692\\"><span class=\\"platform platform-selector PC\\" title=\\"PC (Microsoft Windows)\\">PC</span><span class=\\"platform platform-selector PS4\\" title=\\"PlayStation 4\\">PS4</span><span class=\\"platform platform-selector PS5\\" title=\\"PlayStation 5\\">PS5</span><span class=\\"platform platform-selector XONE\\" title=\\"Xbox One\\">XONE</span><span class=\\"platform platform-selector XSX\\" title=\\"Xbox Series X|S\\">XSX</span></div>
        <div class=\\"game-actions\\"><a href=\\"https://www.igdb.com/games/street-fighter-6\\" target=\\"_blank\\" class=\\"game-details btn btn-igdb\\"><i class=\\"bi bi-info-square\\"></i></a><br><button type=\\"button\\" class=\\"btn btn-outline-danger\\" data-cy=\\"delete-game\\"><i class=\\"bi bi-trash-fill\\"></i></button></div>
      </div>
    </div>
  </div>
</div>
<div class=\\"list-category\\">
  <div class=\\"list-category-heading\\">
    <h5 class=\\"list-heading\\">2024 and beyond</h5><i class=\\"list-icon bi-caret-down-fill\\"></i>
  </div>
  <div id=\\"future\\" class=\\"collapse\\">
    <div class=\\"row\\" data-cy=\\"future\\">
      <div id=\\"game-136627\\" class=\\"col game\\" data-testid=\\"game\\" data-cy=\\"game\\"><img loading=\\"lazy\\" src=\\"//images.igdb.com/igdb/image/upload/t_cover_small/co40fu.jpg\\" class=\\"game-cover\\" data-testid=\\"game-136627-cover\\">
        <div class=\\"game-info\\">
          <h5 data-cy=\\"game-title\\">Suicide Squad: Kill the Justice League</h5>
          <h6>Rocksteady Studios</h6>
          <h6 data-testid=\\"release-date\\" data-cy=\\"release-date\\" class=\\"release-date\\">02/02/2024</h6>
        </div>
        <div class=\\"platforms\\" data-cy=\\"platforms-136627\\"><span class=\\"platform platform-selector PC\\" title=\\"PC (Microsoft Windows)\\">PC</span><span class=\\"platform platform-selector PS5\\" title=\\"PlayStation 5\\">PS5</span><span class=\\"platform platform-selector XSX\\" title=\\"Xbox Series X|S\\">XSX</span></div>
        <div class=\\"game-actions\\"><a href=\\"https://www.igdb.com/games/suicide-squad-kill-the-justice-league\\" target=\\"_blank\\" class=\\"game-details btn btn-igdb\\"><i class=\\"bi bi-info-square\\"></i></a><br><button type=\\"button\\" class=\\"btn btn-outline-danger\\" data-cy=\\"delete-game\\"><i class=\\"bi bi-trash-fill\\"></i></button></div>
      </div>
    </div>
    <div class=\\"row\\" data-cy=\\"future\\">
      <div id=\\"game-136879\\" class=\\"col game\\" data-testid=\\"game\\" data-cy=\\"game\\"><img loading=\\"lazy\\" src=\\"//images.igdb.com/igdb/image/upload/t_cover_small/co6e8b.jpg\\" class=\\"game-cover\\" data-testid=\\"game-136879-cover\\">
        <div class=\\"game-info\\">
          <h5 data-cy=\\"game-title\\">Black Myth: Wukong</h5>
          <h6>Game Science</h6>
          <h6 data-testid=\\"release-date\\" data-cy=\\"release-date\\" class=\\"release-date\\">Q3 2024</h6>
        </div>
        <div class=\\"platforms\\" data-cy=\\"platforms-136879\\"><span class=\\"platform platform-selector PC\\" title=\\"PC (Microsoft Windows)\\">PC</span></div>
        <div class=\\"game-actions\\"><a href=\\"https://www.igdb.com/games/black-myth-wukong\\" target=\\"_blank\\" class=\\"game-details btn btn-igdb\\"><i class=\\"bi bi-info-square\\"></i></a><br><button type=\\"button\\" class=\\"btn btn-outline-danger\\" data-cy=\\"delete-game\\"><i class=\\"bi bi-trash-fill\\"></i></button></div>
      </div>
    </div>
  </div>
</div>
</div></span>
<transition-stub entertoclass=\\"show\\" name=\\"fade\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\"></transition-stub>
</div>"
`;

exports[`renders correctly with released unreleased grouping 1`] = `
"<div class=\\"b-overlay-wrap position-relative\\"><span><div><div><div class=\\"row mt-2\\"><div class=\\"col\\"><div class=\\"input-group\\" role=\\"group\\"><!----><input type=\\"search\\" data-cy=\\"search-games\\" name=\\"search-games\\" class=\\"form-control\\" placeholder=\\"Search Games\\"><div class=\\"btn-group user-menu\\"><button class=\\"btn btn-light dropdown-toggle dropdown-toggle-no-caret\\" type=\\"button\\" id=\\"user-menu\\" data-bs-toggle=\\"dropdown\\" aria-expanded=\\"false\\"> <i class=\\"bi-person-circle\\"></i></button><!----><ul class=\\"dropdown-menu\\" aria-labelledby=\\"user-menu\\" role=\\"menu\\"><li role=\\"presentation\\"><button class=\\"dropdown-item text-danger\\" type=\\"button\\" target=\\"_self\\"><i class=\\"bi-door-open me-2\\"></i>Sign Out </button></li></ul></div></div><!--v-if--></div></div></div><div><ul class=\\"list-actions\\" data-test=\\"list-actions\\"><li id=\\"temporary_game\\" class=\\"\\">Game not found?</li><li id=\\"calendar\\" class=\\"\\">Calendar</li><li id=\\"archive\\" class=\\"\\">Archive</li><li id=\\"grouping\\" class=\\"\\">Grouping</li></ul><div class=\\"list-actions-content\\"><!--v-if--><!--v-if--><!--v-if--><!--v-if--></div></div></div><div><!--v-if--><div class=\\"list-category\\"><div class=\\"list-category-heading\\"><h5 class=\\"list-heading\\">Released Games</h5><i class=\\"list-icon bi-caret-down-fill\\"></i></div><div id=\\"released-games\\" class=\\"collapse\\"><div class=\\"row\\" data-cy=\\"released-games\\"><div id=\\"game-194740\\" class=\\"col game\\" data-testid=\\"game\\" data-cy=\\"game\\"><img loading=\\"lazy\\" src=\\"//images.igdb.com/igdb/image/upload/t_cover_small/co4kxj.jpg\\" class=\\"game-cover\\" data-testid=\\"game-194740-cover\\"><div class=\\"game-info\\"><h5 data-cy=\\"game-title\\">Deliver Us Mars</h5><h6>KeokeN Interactive</h6><h6 data-testid=\\"release-date\\" data-cy=\\"release-date\\" class=\\"release-date\\">02/02/2023</h6></div><div class=\\"platforms\\" data-cy=\\"platforms-194740\\"><span class=\\"platform platform-selector PC\\" title=\\"PC (Microsoft Windows)\\">PC</span><span class=\\"platform platform-selector XONE\\" title=\\"Xbox One\\">XONE</span><span class=\\"platform platform-selector XSX\\" title=\\"Xbox Series X|S\\">XSX</span><span class=\\"platform platform-selector PS4\\" title=\\"PlayStation 4\\">PS4</span><span class=\\"platform platform-selector PS5\\" title=\\"PlayStation 5\\">PS5</span></div>
<div class=\\"game-actions\\"><a href=\\"https://www.igdb.com/games/deliver-us-mars\\" target=\\"_blank\\" class=\\"game-details btn btn-igdb\\"><i class=\\"bi bi-info-square\\"></i></a><br><button type=\\"button\\" class=\\"btn btn-outline-danger\\" data-cy=\\"delete-game\\"><i class=\\"bi bi-trash-fill\\"></i></button></div>
</div>
</div>
<div class=\\"row\\" data-cy=\\"released-games\\">
  <div id=\\"game-191692\\" class=\\"col game\\" data-testid=\\"game\\" data-cy=\\"game\\"><img loading=\\"lazy\\" src=\\"//images.igdb.com/igdb/image/upload/t_cover_small/co5vst.jpg\\" class=\\"game-cover\\" data-testid=\\"game-191692-cover\\">
    <div class=\\"game-info\\">
      <h5 data-cy=\\"game-title\\">Street Fighter 6</h5>
      <h6>Capcom</h6>
      <h6 data-testid=\\"release-date\\" data-cy=\\"release-date\\" class=\\"release-date\\">06/02/2023</h6>
    </div>
    <div class=\\"platforms\\" data-cy=\\"platforms-191692\\"><span class=\\"platform platform-selector PC\\" title=\\"PC (Microsoft Windows)\\">PC</span><span class=\\"platform platform-selector PS4\\" title=\\"PlayStation 4\\">PS4</span><span class=\\"platform platform-selector PS5\\" title=\\"PlayStation 5\\">PS5</span><span class=\\"platform platform-selector XONE\\" title=\\"Xbox One\\">XONE</span><span class=\\"platform platform-selector XSX\\" title=\\"Xbox Series X|S\\">XSX</span></div>
    <div class=\\"game-actions\\"><a href=\\"https://www.igdb.com/games/street-fighter-6\\" target=\\"_blank\\" class=\\"game-details btn btn-igdb\\"><i class=\\"bi bi-info-square\\"></i></a><br><button type=\\"button\\" class=\\"btn btn-outline-danger\\" data-cy=\\"delete-game\\"><i class=\\"bi bi-trash-fill\\"></i></button></div>
  </div>
</div>
</div>
</div>
<div class=\\"list-category\\">
  <div class=\\"list-category-heading\\">
    <h5 class=\\"list-heading\\">Unreleased Games</h5><i class=\\"list-icon bi-caret-down-fill\\"></i>
  </div>
  <div id=\\"unreleased-games\\" class=\\"collapse\\">
    <div class=\\"row\\" data-cy=\\"unreleased-games\\">
      <div id=\\"game-136627\\" class=\\"col game\\" data-testid=\\"game\\" data-cy=\\"game\\"><img loading=\\"lazy\\" src=\\"//images.igdb.com/igdb/image/upload/t_cover_small/co40fu.jpg\\" class=\\"game-cover\\" data-testid=\\"game-136627-cover\\">
        <div class=\\"game-info\\">
          <h5 data-cy=\\"game-title\\">Suicide Squad: Kill the Justice League</h5>
          <h6>Rocksteady Studios</h6>
          <h6 data-testid=\\"release-date\\" data-cy=\\"release-date\\" class=\\"release-date\\">02/02/2024</h6>
        </div>
        <div class=\\"platforms\\" data-cy=\\"platforms-136627\\"><span class=\\"platform platform-selector PC\\" title=\\"PC (Microsoft Windows)\\">PC</span><span class=\\"platform platform-selector PS5\\" title=\\"PlayStation 5\\">PS5</span><span class=\\"platform platform-selector XSX\\" title=\\"Xbox Series X|S\\">XSX</span></div>
        <div class=\\"game-actions\\"><a href=\\"https://www.igdb.com/games/suicide-squad-kill-the-justice-league\\" target=\\"_blank\\" class=\\"game-details btn btn-igdb\\"><i class=\\"bi bi-info-square\\"></i></a><br><button type=\\"button\\" class=\\"btn btn-outline-danger\\" data-cy=\\"delete-game\\"><i class=\\"bi bi-trash-fill\\"></i></button></div>
      </div>
    </div>
    <div class=\\"row\\" data-cy=\\"unreleased-games\\">
      <div id=\\"game-136879\\" class=\\"col game\\" data-testid=\\"game\\" data-cy=\\"game\\"><img loading=\\"lazy\\" src=\\"//images.igdb.com/igdb/image/upload/t_cover_small/co6e8b.jpg\\" class=\\"game-cover\\" data-testid=\\"game-136879-cover\\">
        <div class=\\"game-info\\">
          <h5 data-cy=\\"game-title\\">Black Myth: Wukong</h5>
          <h6>Game Science</h6>
          <h6 data-testid=\\"release-date\\" data-cy=\\"release-date\\" class=\\"release-date\\">Q3 2024</h6>
        </div>
        <div class=\\"platforms\\" data-cy=\\"platforms-136879\\"><span class=\\"platform platform-selector PC\\" title=\\"PC (Microsoft Windows)\\">PC</span></div>
        <div class=\\"game-actions\\"><a href=\\"https://www.igdb.com/games/black-myth-wukong\\" target=\\"_blank\\" class=\\"game-details btn btn-igdb\\"><i class=\\"bi bi-info-square\\"></i></a><br><button type=\\"button\\" class=\\"btn btn-outline-danger\\" data-cy=\\"delete-game\\"><i class=\\"bi bi-trash-fill\\"></i></button></div>
      </div>
    </div>
  </div>
</div>
</div></span>
<transition-stub entertoclass=\\"show\\" name=\\"fade\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\"></transition-stub>
</div>"
`;
